### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by katherinebernstein**

### CODE REVIEW

File: src/main/java/com/dataloom/authorization/HazelcastAuthorizationService.java [hunk 1]
```diff
@@ -24,25 +24,35 @@
...
     public HazelcastAuthorizationService( HazelcastInstance hazelcastInstance, AuthorizationQueryService aqs ) {
         aces = hazelcastInstance.getMap( HazelcastMap.PERMISSIONS.name() );
         this.aqs = checkNotNull( aqs );
+        this.executor = hazelcastInstance.getDurableExecutorService( "default" );
     }
-
+    
+    private void submitNewPermissionsToElasticsearch( List<AclKeyPathFragment> key, Principal principal ) {
+    	Set<Permission> permissions = aces.get( new AceKey( key, principal ) );
+    	// executor.submit ... updateEntitySetAclsInElasticsearch( entitySetId, principal, permissions)
```
Comment (geekbeast) [inline src/main/java/com/dataloom/authorization/HazelcastAuthorizationService.java]: Cleanup comment code.

File: src/main/java/com/kryptnostic/conductor/rpc/ConductorSparkApi.java [hunk 2]
```diff
@@ -31,4 +40,19 @@ QueryResult getAllEntitiesOfEntitySet(
      * @return QueryResult of UUID's matching the lookup request
      */
     QueryResult getFilterEntities( LookupEntitiesRequest request );
+
+	Boolean submitEntitySetToElasticsearch( EntitySet entitySet, List<PropertyType> propertyTypes, Principal principal );
```
Comment (geekbeast) [inline src/main/java/com/kryptnostic/conductor/rpc/ConductorSparkApi.java]: These should be in their own interface. For now it's fine if `ConductorSparkImpl` implements both interfaces.

File: src/main/java/com/dataloom/authorization/HazelcastAuthorizationService.java (hunk 1 above)
Comment (geekbeast) [inline src/main/java/com/dataloom/authorization/HazelcastAuthorizationService.java]: This function doesn't see to do anything currently.

File: src/main/java/com/dataloom/authorization/HazelcastAuthorizationService.java [hunk 3]
```diff
@@ -24,25 +24,35 @@
...
+    	// entitySet.permissions.<principal.type>
+    }
+    
     @Override
     public void addPermission(
             List<AclKeyPathFragment> key,
             Principal principal,
             Set<Permission> permissions ) {
         aces.executeOnKey( new AceKey( key, principal ), new PermissionMerger( permissions ) );
+        submitNewPermissionsToElasticsearch( key, principal );
```
Comment (geekbeast) [inline src/main/java/com/dataloom/authorization/HazelcastAuthorizationService.java]: I would remove all these one off calls to submit new permissions here and instead use a latch, hooking the controller update call to post an AclData object the event bus, and then have a ElastichSearchUpdateService that listens for events.

File: src/main/java/com/kryptnostic/conductor/rpc/Lambdas.java [hunk 4]
```diff
@@ -33,4 +41,28 @@ public static Runnable foo() {
         return (Function<ConductorSparkApi, QueryResult> & Serializable) ( api ) -> api
                 .getAllEntitiesOfEntitySet( entityFqn, entitySetName, authorizedProperties );
     }
+    
+    public static Function<ConductorSparkApi, Boolean> submitEntitySetToElasticsearch( EntitySet entitySet, List<PropertyType> propertyTypes, Principal principal ) {
```
Comment (geekbeast) [inline src/main/java/com/kryptnostic/conductor/rpc/Lambdas.java]: This is getting pretty ugly (due to past choices) we may have to modify how  we are grouping these together separate from this PR.

File: src/main/java/com/kryptnostic/conductor/rpc/ConductorSparkApi.java (hunk 2 above)
Comment (katherinebernstein) [inline src/main/java/com/kryptnostic/conductor/rpc/ConductorSparkApi.java]: yeah that makes a lot of sense. I'll file a ticket for myself on this

File: src/main/java/com/kryptnostic/conductor/rpc/Lambdas.java (hunk 4 above)
Comment (katherinebernstein) [inline src/main/java/com/kryptnostic/conductor/rpc/Lambdas.java]: cool, yeah it is pretty ugly

File: src/main/java/com/kryptnostic/datastore/services/EdmService.java [hunk 5]
```diff
@@ -202,6 +209,8 @@ public void deleteEntitySet( UUID entitySetId ) {
                 .forEach( authorizations::deletePermissions );
 
         Util.deleteSafely( entitySets, entitySetId );
+        executor.submit( ConductorCall
```
Comment (geekbeast) [inline src/main/java/com/kryptnostic/datastore/services/EdmService.java]: This should follow the same Guava event bus mechanics.

File: src/main/java/com/kryptnostic/conductor/rpc/ConductorElasticsearchApi.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import com.dataloom.authorization.Permission;
+import com.dataloom.authorization.Principal;
+import com.dataloom.edm.internal.EntitySet;
+import com.dataloom.edm.internal.PropertyType;
+import com.google.common.base.Optional;
+
+public interface ConductorElasticsearchApi {
+	
+	// index setup consts
+	final String ENTITY_SET_DATA_MODEL = "entity_set_data_model";
```
Comment (geekbeast) [inline src/main/java/com/kryptnostic/conductor/rpc/ConductorElasticsearchApi.java]: Do any of these properties get passed as function arguments anywhere?

Comment (katherinebernstein) [inline src/main/java/com/kryptnostic/conductor/rpc/ConductorElasticsearchApi.java]: yep, in ConductorElasticsearchImpl in kindling

File: src/main/java/com/kryptnostic/datastore/services/EdmService.java (hunk 5 above)
Comment (katherinebernstein) [inline src/main/java/com/kryptnostic/datastore/services/EdmService.java]: updated!

**Pull request merged**